client/js/shout.js

```diff
@@ -600,7 +613,9 @@ $(function() {
 		var button = sidebar.find(".chan[data-target=" + target + "]");
 		var isQuery = button.hasClass("query");
 		var type = msg.type;
-		var highlight = type.contains("highlight");
+		var highlight = type.contains("highlight")|| highlights.some(function(h) {
```

Comment:
AFAICS, the type is sent to us from slate, so it'll just be an indiation that the username was mentioned, I assume. I'll double check later, though.

I guess we could set the type earlier on if it fits a custom highlight. That might work well. But we'll obviously still need to do this check somewhere in this function.